### PR TITLE
Use poshlib

### PR DESCRIPTION
**Pull request by andrewgdotcom**
No rebasing, conflict resolution was enough.

### CODE REVIEW

**Review by GMDSantana — APPROVED** [review]

File: constructor.sh [hunk 1]
```diff
@@ -108,7 +108,6 @@ echo DEBUG=$DEBUG
 # We can optionally override the branches of our repo dependencies
 # But most of the time we probably just want "master"
 : "${GIT_DEMOS_BRANCH:=master}" # if it is unset or empty therefore the initial ':'
```
Comment (itudoben) [inline constructor.sh:110]: I like this syntax, so powerful!

File: tests/test-gradle [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+tmpdir=$(mktemp -d)
+tmpfile=$(mktemp)
+
+cd "$tmpdir"
+git clone --recurse-submodules git@github.com:sirensolutions/siren-platform
+cd siren-platform
+
+gradleOpts=(-p benchmark -is \
+    -Pfederate.commit=8579c77702b13bc826e651db940b88d95bea76e5 \
```
Comment (itudoben) [inline tests/test-gradle]: Maybe this commit could be a parameter with a default value as this depends on the ES version specified above (right now 7.9.2).

File: tests/test-gradle [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    -PartifactoryApiKey="$ARTIFACTORY_API_KEY" \
+    -PgithubCredentials="$GITHUB_CREDENTIALS" \
```
Comment (itudoben) [inline tests/test-gradle]: This should not be necessary as long as `GITHUB_CREDENTIALS` is set.
```suggestion
```

**Review by itudoben — APPROVED** [review]
Looks good. A little change not mandatory though.

File: tests/test-gradle (hunk 2 above)
Comment (andrewgdotcom) [reply tests/test-gradle]: done

File: constructor.sh (hunk 1 above)
Comment (andrewgdotcom) [reply constructor.sh:110]: :sunglasses:

**Review by GMDSantana — APPROVED** [review]

**Pull request merged**